anywidget/_descriptor.py

```diff
@@ -635,3 +643,17 @@ def _get_pydantic_state(obj: pydantic.BaseModel) -> dict:
     expects.)
     """
     return json.loads(obj.json())
+
+# ------------- msgspec support --------------
+
+def _is_msgspec_struct(obj: Any) -> TypeGuard[msgspec.Struct]:
+    """Return `True` if an object is an instance of msgspec.Struct."""
+    msgspec = sys.modules.get("msgspec")
+    return isinstance(obj, msgspec.Struct) if msgspec is not None else False
+
+
+def _get_msgspec_state(obj: msgspec.Struct) -> dict:
+    """Get the state of a msgspec.Struct instance."""
+    import msgspec
+
+    return msgspec.json.decode(msgspec.json.encode(obj))
```

Comment:
particularly painful here...  but we'll address this in a followup

Reply:
It's not clear to me if/why this is necessary, but if it is it'd be faster to do `msgspec.to_builtins(obj)` instead. This should have the same end result, but avoid encoding then decoding.

Reply:
Thanks @jcrist 

The reason this is here (temporarily, as a hack, sadly, for now)... is that we don't yet override the actual Comm object's serialization event.  Comm comes from another library, and expects a dict object and serializes with the stdlib json.  And since json doesn't provide an easy way to define serialization of custom classes _on the class_ (one must customize the serializer itself), here we just try to make sure that the dict will be "plain enough" for json.

This pattern here mostly came from the pydantic pattern used above, and it was earmarked here for later investigation.  

But thank you for pointing out to builtins!  That's great